### PR TITLE
fix: break path w/ <wbr>s to avoid copying ZWSPs

### LINKED ISSUE #7513: ZERO WIDTH SPACE (U+200B) in path copied if selection is not exact
### Q&A (please complete the following information)
 - OS: Win 10
 - Browser: Firefox
 - Version: 92.0
 - Method of installation: `dist/`
 - Swagger-UI version: v3.24.2
 - Swagger/OpenAPI version: [e.g. Swagger 2.0, OpenAPI 3.0]

### Describe the bug you're encountering

If you drag to select the path, but select a little bit too far, the zero-width space removal doesn't happen, and you copy a bunch of zero-width spaces into your code to cause invisible bugs.

### To reproduce...

1. Start selecting the path text from right to left.
2. Drag slightly further than the start of the text.
3. Copy

### Screenshots

image
Will copy zero width spaces (on left and in middle of path).

If you very carefully stop at the start of the text things will be OK:
image
(OK)

### Cause

Caused by https://github.com/swagger-api/swagger-ui/pull/5184

The PR above injects zero width spaces into the path, but includes an `onCopyCapture` function which is triggered when the user copies, which removes all zero width spaces from the copied text. However that event is only triggered if the selection is _only on the path `<span>`_. In reality, users will often drag a little further to make sure all the characters are selected, but this (for some reason) means the `onCopyCapture` is not triggered and ZERO WIDTH SPACEs are copied.

### Fix
`<wbr>` should be used instead, which requires no `onCopyCapture` faffing: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/wbr

### PR DESCRIPTION
**Pull request by MingweiSamuel**
### Description
- Use <wbr> instead of ZERO-WIDTH SPACE (U+200B) to break segments in
  operation-summary-path.jsx
- Remove no-longer-needed onCopyCapture listener which previously
  stripped ZWSPs

### Motivation and Context
Fixes #7513

### How Has This Been Tested?
Basic local test, HTML looks right and behaves the same as before, just without copying of ZWSPs.

### Screenshots (if appropriate):
N/A

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/operation-summary-path.jsx [hunk 1]
```diff
@@ -34,17 +28,22 @@ export default class OperationSummaryPath extends PureComponent{
       isDeepLinkingEnabled,
     } = operationProps.toJS()
 
+    // Add <wbr> word-break elements between each segment, before the slash.
+    const textWithBreaks = path.split(/(?=\/)/g)
```
Comment (MingweiSamuel) [inline src/core/components/operation-summary-path.jsx:32]: Removed the semicolon just now

File: src/core/components/operation-summary-path.jsx [hunk 2]
```diff
@@ -34,17 +28,22 @@ export default class OperationSummaryPath extends PureComponent{
...
-      <span className={ deprecated ? "opblock-summary-path__deprecated" : "opblock-summary-path" } 
-        onCopyCapture={this.onCopyCapture}
+      <span className={ deprecated ? "opblock-summary-path__deprecated" : "opblock-summary-path" }
         data-path={path}>
         <DeepLink
             enabled={isDeepLinkingEnabled}
             isShown={isShown}
             path={createDeepLinkPath(`${tag}/${operationId}`)}
-            text={path.replace(/\//g, "\u200b/")} />
+            text={textWithBreaks} />
```
Comment (char0n) [inline src/core/components/operation-summary-path.jsx:46]: [DeepLink.text expects a string](https://github.com/swagger-api/swagger-ui/blob/d2e931f3e1c410da92bcad035d200d9d9ea74a80/src/core/components/deep-link.jsx#L17), so prop validation will fail on this change. `textWithBreaks` is a list of strings and react components.

We have couple of options how to resolve it:

1. allow DeepLink to accept text prop in following type: `string` | [`string` | `React Component`]


2. allow DeepLink to accept text prop in following type: `string` | `React Component`

I think I like second option more as it defines simple prop type and we can use following code:

```jsx
const pathWithBreaks = <>{pathParts}</>;

text={pathWithBreaks} />
```


Can you think of anything better?

**Review by char0n — CHANGES_REQUESTED** [review]
Thanks for the PR!

You did a great job. Can you please look at my code review comments before we move forward?

File: src/core/components/operation-summary-path.jsx (hunk 2 above)
Comment (MingweiSamuel) [reply src/core/components/operation-summary-path.jsx:46]: Second option seems reasonable to me (but I am not experienced with React or JSX).

Comment (MingweiSamuel) [reply src/core/components/operation-summary-path.jsx:46]: Based on https://reactjs.org/docs/typechecking-with-proptypes.html seems like `PropTypes.node` is what we want, which accepts anything renderable, including both arrays & elements, and strings:
```js
// Anything that can be rendered: numbers, strings, elements or an array
// (or fragment) containing these types.
optionalNode: PropTypes.node,
```

Alternative is `PropTypes.oneOf([ PropTypes.string, PropTypes.element ])` which would be exactly `2.` above, I think.

I'll do the former for now, and keep the array.

**Comment by MingweiSamuel** [comment]
Seems there is a warning I need to fix
```
Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `_class4`. It was passed a child from OperationSummaryPath. See https://fb.me/react-warning-keys for more information.
```
Probably a pretty basic react thing

**Comment by MingweiSamuel** [comment]
Replacing
```js
pathParts.splice(i, 0, <wbr />)
```
with
```js
pathParts.splice(i, 0, <wbr key={i} />)
```
Fixed the above. Ready for another review.

File: src/core/components/operation-summary-path.jsx (hunk 2 above)
Comment (char0n) [reply src/core/components/operation-summary-path.jsx]: PropTypes.node looks like an idiomatic thing for this case. Good catch!

File: src/core/components/operation-summary-path.jsx [hunk 3]
```diff
@@ -34,17 +28,22 @@ export default class OperationSummaryPath extends PureComponent{
       isDeepLinkingEnabled,
     } = operationProps.toJS()
 
+    // Add <wbr> word-break elements between each segment, before the slash.
+    const textWithBreaks = path.split(/(?=\/)/g)
+    for (let i = 1; i < textWithBreaks.length; i += 2) {
+      textWithBreaks.splice(i, 0, <wbr />)
```
[the remark this answers is not in the document]
Comment (char0n) [reply src/core/components/operation-summary-path.jsx]: The key was added.

**Review by char0n — APPROVED** [review]
We're looking good here. Thank you for your contribution!

File: src/core/components/operation-summary-path.jsx [hunk 4]
```diff
@@ -34,17 +28,22 @@ export default class OperationSummaryPath extends PureComponent{
       isDeepLinkingEnabled,
     } = operationProps.toJS()
 
+    // add <wbr> word-break elements between each segment, before the slash
```
Comment (char0n) [inline src/core/components/operation-summary-path.jsx:31]: Just nitpick here. Do we want to provide more descriptive explanation why this is here?
Here is my initial suggetion.

```suggestion
    /**
     * Add <wbr> word-break elements between each segment, before the slash
     * to allow browsers an opportunity to break long paths into sensible segments.
     */
```

**Comment by MingweiSamuel** [comment]
Thanks! 🎉
